posthog/queries/util.py

```diff
@@ -40,11 +42,20 @@ def parse_timestamps(filter: FilterType, team_id: int, table: str = "") -> Tuple
     return date_from or "", date_to or "", params
 
 
-def format_ch_timestamp(timestamp: datetime, filter, default_hour_min: str = " 00:00:00"):
+def format_ch_timestamp(
+    timestamp: datetime, filter, default_hour_min: str = " 00:00:00", timezone: Optional[str] = None
+):
+    if timezone:
+        # Here we probably get a timestamp set to the beginning of the day (00:00), in UTC
+        # We need to convert that UTC timestamp to the local timestamp (00:00 in US/Pacific for example)
+        # Then we convert it back to UTC (08:00 in UTC)
+        timestamp = pytz.timezone(timezone).localize(timestamp.replace(tzinfo=None)).astimezone(pytz.UTC)
```

Comment:
are we 100% the timestamps coming from user args are always UTC? Seems more robust to check this somehow or make sure it's always explicit from the user

Reply:
Quick grep through existing insights suggests this has never happened yet from the frontend, but could see this happening from someone calling the api. Have added an exception if this does happen.